reckon-gradle/src/main/java/org/ajoberstar/reckon/gradle/ReckonPlugin.java

```diff
@@ -37,14 +44,57 @@ public void apply(Project project) {
             "org.ajoberstar.grgit",
             plugin -> {
               Grgit grgit = (Grgit) project.findProperty("grgit");
-              extension.setVcsInventory(extension.git(grgit));
+              if (grgit != null) {
+                extension.setVcsInventory(extension.git(grgit));
+              }
             });
 
     DelayedVersion sharedVersion = new DelayedVersion(extension::reckonVersion);
     project.allprojects(
         prj -> {
           prj.setVersion(sharedVersion);
         });
+
+    Task tag = createTagTask(project, extension);
+    Task push = createPushTask(project, extension);
+    push.dependsOn(tag);
+  }
+
+  private Task createTagTask(Project project, ReckonExtension extension) {
+    Task task = project.getTasks().create(TAG_TASK);
+    task.setDescription("Tag version inferred by reckon.");
+    task.setGroup("publish");
+    task.onlyIf(
+        t -> {
+          // using the presence of build metadata as the indicator of taggable versions
+          return !project.getVersion().toString().contains("+");
+        });
+    task.doLast(
+        t -> {
+          Map<String, Object> args = new HashMap<>();
+          args.put("name", project.getVersion());
+          args.put("message", project.getVersion());
+          extension.getGrgit().getTag().add(args);
+        });
+    return task;
+  }
+
+  private Task createPushTask(Project project, ReckonExtension extension) {
+    Task task = project.getTasks().create(PUSH_TASK);
+    task.setDescription("Push version tag created by reckon.");
+    task.setGroup("publish");
+    task.onlyIf(
+        t -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement and then remove useless return keyword [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.ajoberstar.com/coding_rules#rule_key=squid%3AS1602)